public-ui/src/app/data-browser/views/pm/pm-react.component.tsx

```diff
@@ -0,0 +1,488 @@
+        male = g;
+      } else if (g.stratum2 === '8532') {
+        female = g;
+      } else if (g.stratum2 === '0') {
+        other = g;
+      }
+    }
+
+    // Order genders how we want to display  Male, Female , Others
+    if (male) { results.push(male); }
+    if (female) { results.push(female); }
+    if (other) { results.push(other); }
+    analysis.results = results;
+  }
+
+  setAnalysisStratum(results: any) {
+    for (const r of results) {
+      if (r.analysisStratumName === null) {
+        r.analysisStratumName = GENDER_STRATUM_MAP[r.stratum3];
+      }
+    }
+  }
+
+  arrangeConceptAnalyses(concept: any) {
+    if (concept.analyses.genderAnalysis) {
+      this.organizeGenders(concept);
+    }
+
+    let genders = ['8507', '8532', '0'];
+    let prevResult;
+    for (const gca of concept.analyses.measurementGenderCountAnalysis) {
+      if (gca.results.length < 3) {
+        for (const result of gca.results) {
+          prevResult = result;
+          genders = genders.filter(item => item !== result.stratum3);
+        }
+        for (const gender of genders) {
+          const missingResult = { ...prevResult };
+          missingResult.stratum3 = gender;
+          missingResult.countValue = 20;
+          missingResult.sourceCountValue = 20;
+          gca.results.push(missingResult);
+        }
+      }
+      this.setAnalysisStratum(gca.results);
+    }
+  }
+
+  showMeasurement(pmConceptGroup: any, concept: any) {
+    this.setUnit(concept);
+    this.setState({selectedGroup: pmConceptGroup, selectedConcept: concept});
+  }
+
+  setUnit(concept: any) {
+    const unitNames = [];
+    if (concept.analyses && concept.analyses.measurementGenderCountAnalysis) {
+      for (const r of concept.analyses.measurementGenderCountAnalysis) {
+        let tempUnitNames = r.results.map(({ stratum2 }) => stratum2);
+        tempUnitNames = tempUnitNames.filter((elem, index, self) => index === self.indexOf(elem));
+        unitNames.push(...tempUnitNames);
+      }
+    }
+    if (unitNames.length > 0) {
+      this.setState({selectedConceptUnit: unitNames[0], unitNames: unitNames}, () => {
+        this.setAnalysis();
+      });
+      return unitNames[0];
+    }
+    return null;
+  }
+
+  setConceptUnit(unit) {
+    this.setState({selectedConceptUnit: unit}, () => {
+      this.setAnalysis();
+    });
+  }
+
+
+
+  setAnalysis() {
+    const {selectedConcept, selectedConceptUnit} = this.state;
+    if (['903120', '903111'].indexOf(selectedConcept.conceptId) === -1) {
+      let temp = selectedConcept.analyses.measurementValueGenderAnalysis.filter(
+        a => a.unitName.toLowerCase() ===
+          selectedConceptUnit.toLowerCase());
+      this.setState({selectedConceptValueAnalysis: temp[0]});
+      temp = selectedConcept.analyses.measurementGenderCountAnalysis.filter(
+        a => a.unitName.toLowerCase() ===
+          selectedConceptUnit.toLowerCase());
+      this.setState({selectedConceptValueCountAnalysis: temp[0]});
```

Comment:
Nit: Can combine these into a single `setState`:
```suggestion
      this.setState({
        selectedConceptValueAnalysis: selectedConcept.analyses.measurementValueGenderAnalysis.filter(
          a => a.unitName.toLowerCase() === selectedConceptUnit.toLowerCase())[0],
        selectedConceptValueCountAnalysis: selectedConcept.analyses.measurementGenderCountAnalysis.filter(
          a => a.unitName.toLowerCase() === selectedConceptUnit.toLowerCase())[0]
      });
```

Reply:
Thanks. changed this.